### PR TITLE
Miners now get planet-limited night vision goggles

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
rscadd: Miners now get planet-limited night vision goggles instead of mesons.
/:cl:

Kor suggestion here:
https://github.com/tgstation/tgstation/pull/28764#issuecomment-310863057

On-station they're as effective as mesons are on mining.

### CODE REVIEW

File: code/modules/clothing/glasses/glasses.dm [hunk 1]
```diff
@@ -129,6 +129,54 @@
 	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_INVISIBLE
 	glass_colour_type = /datum/client_colour/glass_colour/green
 
+/obj/item/clothing/glasses/night/mining
+	name = "Planetary Night Vision Goggles"
```
Comment (Xhuis) [inline code/modules/clothing/glasses/glasses.dm]: Should this be lowercase?

File: code/modules/clothing/glasses/glasses.dm [hunk 2]
```diff
@@ -129,6 +129,54 @@
 	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_INVISIBLE
 	glass_colour_type = /datum/client_colour/glass_colour/green
 
+/obj/item/clothing/glasses/night/mining
+	name = "Planetary Night Vision Goggles"
+	desc = "Used by mining staff to see in the dark; however, it is an extremely cheap brand that automatically shuts off in the presence of "
```
Comment (Xhuis) [inline code/modules/clothing/glasses/glasses.dm]: In the presence of...

File: code/modules/clothing/glasses/glasses.dm (hunk 1 above)
Comment (ChangelingRain) [inline code/modules/clothing/glasses/glasses.dm]: none of the other things are but probably _technically_ yes
I don't feel like updating all of them though

File: code/modules/clothing/glasses/glasses.dm (hunk 2 above)
Comment (ChangelingRain) [inline code/modules/clothing/glasses/glasses.dm]: see Initialize()

Comment (Xhuis) [inline code/modules/clothing/glasses/glasses.dm]: I can read I swear

File: code/modules/clothing/glasses/glasses.dm (hunk 1 above)
Comment (Xhuis) [inline code/modules/clothing/glasses/glasses.dm]: Then my people need me

File: code/modules/clothing/glasses/glasses.dm [hunk 3]
```diff
@@ -129,6 +129,54 @@
 	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_INVISIBLE
 	glass_colour_type = /datum/client_colour/glass_colour/green
 
+/obj/item/clothing/glasses/night/mining
+	name = "Planetary Night Vision Goggles"
+	desc = "Used by mining staff to see in the dark; however, it is an extremely cheap brand that automatically shuts off in the presence of "
+	origin_tech = "materials=2;magnets=1;engineering=2"
+	darkness_view = 2
+	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_VISIBLE
+	var/static/list/nvg_station_failure_excuses = list("cosmic rays", "artificial gravity")
```
Comment (ChangelingRain) [inline code/modules/clothing/glasses/glasses.dm]: additional bad jokes required

File: code/modules/clothing/glasses/glasses.dm [hunk 4]
```diff
@@ -129,6 +129,54 @@
...
+	origin_tech = "materials=2;magnets=1;engineering=2"
+	darkness_view = 2
+	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_VISIBLE
+	var/static/list/nvg_station_failure_excuses = list("cosmic rays", "artificial gravity")
+	var/picked_excuse
+	var/nvg_on = FALSE
+
+/obj/item/clothing/glasses/night/mining/Initialize()
+	. = ..()
+	picked_excuse = pick(nvg_mining_failure_excuses)
```
Comment (lzimann) [inline code/modules/clothing/glasses/glasses.dm]: You declared as nvg_station and is using nvg_mining here.

File: code/modules/clothing/glasses/glasses.dm [hunk 5]
```diff
@@ -129,6 +129,54 @@
...
+			to_chat(C, "<span class='warning'>Due to the presence of [picked_excuse], your [name] have disabled themselves to prevent damage.</span>")
+		if(C.glasses == src)
+			C.update_sight()
+
+/obj/item/clothing/glasses/night/mining/process()
+	var/turf/T = get_turf(src)
+	if(T && T.z == ZLEVEL_MINING)
+		if(!nvg_on) //if we're on mining, turn nvg ON
+			toggle_mode(loc)
+	else if(mvg_on) //otherwise, if we're not on mining, turn nvg back OFF
```
Comment (lzimann) [inline code/modules/clothing/glasses/glasses.dm]: MVG

File: code/modules/clothing/glasses/glasses.dm (hunk 3 above)
Comment (ike709) [inline code/modules/clothing/glasses/glasses.dm]: the neutron flow

File: code/modules/clothing/glasses/glasses.dm [hunk 6]
```diff
@@ -129,6 +129,54 @@
 	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_INVISIBLE
 	glass_colour_type = /datum/client_colour/glass_colour/green
 
+/obj/item/clothing/glasses/night/mining
+	name = "Planetary Night Vision Goggles"
+	desc = "Used by mining staff to see in the dark; however, it is an extremely cheap brand that automatically shuts off in the presence of "
+	origin_tech = "materials=2;magnets=1;engineering=2"
+	darkness_view = 2
+	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_VISIBLE
+	var/static/list/nvg_station_failure_excuses = list("electromagnetic radiation", "artificial gravity", "strange signals", "cosmic rays", "dark matter")
```
Comment (ChangelingRain) [inline code/modules/clothing/glasses/glasses.dm]: jokes, jokes, grant us jokes

Comment (ike709) [inline code/modules/clothing/glasses/glasses.dm]: neutron flows

It's always nice adding Doctor Who references.

Comment (XDTM) [inline code/modules/clothing/glasses/glasses.dm]: excessive surveillance

**Comment by Kromgar** [comment]
Fullbright please?

**Comment by ChangelingRain** [comment]
It's the same brightness as standard night vision goggles.

**Comment by SgtxRootbeer** [comment]
nah, just give them mesons.

**Comment by ShadowDeath6** [comment]
Giving them NVG is a slight buff versus fauna, since they have full view without needing lights rather than the pseudo vision that mesons give. 

Probably won't matter.

**Comment by KorPhaeron** [comment]
It is a pretty huge buff vs ghost roles

Slight buff against fauna because you can see them from 7 tiles away now when they usually can only see around 3 or 4

**Comment by Repukan** [comment]
Just bring back mesons.

I don't understand why we need a fucking compromise when the reason for meson removal was never added (or worked on as far as I know.)

**Comment by Kromgar** [comment]
j want fullbright because  its fucking near impossible to see anything in ash storms. Used to be it was easy to see shit

**Comment by Kromgar** [comment]
Onlg things easy to see are bubblegum and watchers

**Comment by Repukan** [comment]
Go to an oculist dude geez

**Comment by tortellinitony** [comment]
Why can't you just make mesons work differently in both environments, rather than adding another snowflake item?

**Comment by Kromgar** [comment]
@Repukan I'm not turning up brightness for one game.

**Comment by KorPhaeron** [comment]
If secretive ruins ever get added we can revisit this but nobody has bothered in the last ~2 months

**Pull request closed without merging**